### PR TITLE
fix: allReservation filters

### PR DESCRIPTION
**Pull request by vincit-matu**
- updates queries and gql-types + codegen

## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Adds some filters, and re-arranges them on the all-reservations page in the admin UI.
- Adds conversion functions to fix flaky API typing (`ReservationStateChoice` & `ReservationTypeChoice`), which can(/should) be removed once the API is up to date and codegen gives correct typing

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Go to "Kaikki varaukset" in the admin view, expand the search-filters to show them all, and test them out

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- TILA-3401

### CODE REVIEW

File: apps/admin-ui/src/component/reservations/Filters.tsx [hunk 1]
```diff
@@ -10,14 +9,21 @@ import {
...
 import { useSearchParams } from "react-router-dom";
 import { SearchTags } from "../SearchTags";
-import { OrderStatus, State } from "@gql/gql-types";
+import {
+  OrderStatusWithFree,
+  ReservationTypeChoice,
+  State,
+} from "@gql/gql-types";
 import { fromUIDate, isValidDate } from "common/src/common/util";
+import { Select } from "@/component/Select";
```
Comment (joonatank) [inline apps/admin-ui/src/component/reservations/Filters.tsx:26]: dont't use this component import directly from hds

File: apps/admin-ui/src/component/reservations/Filters.tsx [hunk 2]
```diff
@@ -132,34 +153,53 @@ export function Filters({
...
+            onChange={(val) => {
+              const params = new URLSearchParams(searchParams);
+              if (val != null) {
+                params.set("recurring", val.toString());
+              } else params.delete("recurring");
+              setParams(params, { replace: true });
+            }}
+            value={new URLSearchParams(searchParams).get("recurring") ?? ""}
+            clearable
+          />
```
Comment (joonatank) [inline apps/admin-ui/src/component/reservations/Filters.tsx:201]: Separate component similar to others.

File: apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx [hunk 3]
```diff
@@ -16,20 +16,24 @@ import { fromUIDate, toApiDate } from "common/src/common/util";
 import { filterNonNullable, toNumber } from "common/src/helpers";
 import { useSearchParams } from "react-router-dom";
 
-function transformPaymentStatusSafe(t: string): OrderStatus | null {
+function transformPaymentStatusSafe(
+  t: string
+): OrderStatusWithFree | "free" | null {
```
Comment (joonatank) [inline apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx]: `"free"` is unnecessary

File: packages/common/src/conversion.ts [hunk 4]
```diff
@@ -38,3 +44,62 @@ export function convertWeekday(d: Weekday): Day {
...
+    case "BLOCKED":
+      return Type.Blocked;
+    case "NORMAL":
+      return Type.Normal;
+    case "SEASONAL":
+      return Type.Seasonal;
+    case "STAFF":
+      return Type.Staff;
+    default:
+      return;
```
Comment (joonatank) [inline packages/common/src/conversion.ts:61]: unnecessary `return`

**Review by joonatank — CHANGES_REQUESTED** [review]

File: apps/admin-ui/src/component/reservations/Filters.tsx (hunk 2 above)
Comment (vincit-matu) [reply apps/admin-ui/src/component/reservations/Filters.tsx]: Good point! Extracted to it's own component (`SelectFilter.tsx`)

File: apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx (hunk 3 above)
Comment (vincit-matu) [reply apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx]: That's weird, I'm sure I deleted "free" as a return type from that already, and my local branch doesn't show it.. 😮

Comment (vincit-matu) [reply apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx]: ..fixed anyhow now - not that I actually changed anything 😅

**Review by joonatank — APPROVED** [review]
👍

**Review by joonatank — CHANGES_REQUESTED** [review]
Blocking because the backend branch had types fixed.

Should remove the converters and update types before merge.

**Review by joonatank — APPROVED** [review]

**Pull request merged**